### PR TITLE
remove docs for spark2

### PR DESCRIPTION
**Pull request by johnnyaug**
Part of #5107

### CODE REVIEW

**Comment by rmoff** [comment]
I don't have sufficient context/knowledge to comment on the removal of support for Spark 2, so will leave that to other reviewers. 
In terms of the docs should we call out that Spark 2 is not supported (vs just omitting mention of it)?

**Review by rmoff — CHANGES_REQUESTED** [review]
In terms of the docs should we call out that Spark 2 is not supported (vs just omitting mention of it)?

**Comment by johnnyaug** [comment]
@rmoff, done.

**Review by arielshaqed — APPROVED** [review]
Neat, thanks!

Please open an issue to discuss whether we should get rid of the "2.12" flavour of the JAR: it might no longer be needed!  (But not clear because maybe we'll need it in future, when _another_ Spark version bumps their Scala runtime version???)

File: docs/reference/spark-client.md [hunk 1]
```diff
@@ -19,25 +19,16 @@ Utilize the power of Spark to interact with the metadata on lakeFS. Possible use
 
 ## Getting Started
 
+We are removing support for Spark 2 in the lakeFS metadata client.
+{: .note }
+
```
Comment (rmoff) [inline docs/reference/spark-client.md:24]:
```suggestion
Please note that Spark 2 is no longer supported with the lakeFS metadata client.
{: .note }

```

File: clients/spark/README.md [hunk 2]
```diff
@@ -7,21 +7,15 @@ Read metadata from lakeFS into Spark.
 1. Read Graveler meta-ranges, ranges and entries.
 1. Export data from lakeFS to any object storage (see [docs](https://docs.lakefs.io/reference/export.html)).
 
-## Scala/Spark compatibility
-
-Two versions are available for the client, compatible with the following Spark/Scala versions:
-1. Spark 2 / Scala 2.11
-1. Spark 3 / Scala 2.12
-   
 ## Installation
```
Comment (rmoff) [inline clients/spark/README.md:10]:
```suggestion
Please note that Spark 2 is no longer supported with the lakeFS client.
{: .note }

## Installation
```

**Review by rmoff — CHANGES_REQUESTED** [review]
i've added some suggested wording

**Pull request merged**